lib/util/url.ts

```diff
@@ -1,3 +1,16 @@
+import { parse } from 'url';
+import urlJoin from 'url-join';
+
 export function ensureTrailingSlash(url: string): string {
   return url.replace(/\/?$/, '/');
 }
+
+export function resolveBaseUrl(
+  baseUrl: string,
+  pathOrUrl: string | URL
+): string {
+  const pathOrUrlString = pathOrUrl.toString();
+  const { host, path } = parse(pathOrUrlString);
```

Comment:
```suggestion
  const { host, path } = new URL(pathOrUrlString);
```

`URL` is on global object.